### PR TITLE
Add IntelliJ IDEA Java code style settings

### PR DESCRIPTION
**Pull request by mandybess**
Code styles for Java! Yay! Also included two install scripts to easily import and use the style in Android Studio.

These are not final by any means, just a starting place. Open to feedback, ideas, areas of improvement!

It will be cool to reference these once we open source some android projects, so anyone who wants to contribute can do so easily and with the correct styles so the code is consistent and clean :+1:

### CODE REVIEW

File: style/android/java/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Java Code Styles
+================
+
+IntelliJ IDEA code style settings for Thoughtbot's Android projects.
```
Comment (jferris) [inline style/android/java/README.md]: We prefer to spell thoughtbot with a lowercase t

File: style/android/java/install.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# Installs Thoughtbot's IntelliJ configs into your user configs.
```
Comment (gabebw) [inline style/android/java/install.sh]: We leave thoughtbot lowercase, even when it starts a sentence, even when it makes a whole bunch of extra work for us over the course of _years_

File: style/android/java/install.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+echo "Installing Thoughtbot IntelliJ configs..."
+
+CONFIGS="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/configs"
```
Comment (gabebw) [inline style/android/java/install.sh]: Could this be:

```
CONFIGS=$(dirname "${BASH_SOURCE[0]}")/configs
```

File: style/android/java/install.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+for i in $HOME/Library/Preferences/IntelliJIdea*  \
```
Comment (gabebw) [inline style/android/java/install.sh]: It's probably a good idea to put `$HOME` in quotes: `"$HOME"/Library/Preferences/IntelliJIde`

File: style/android/java/install.sh (hunk 2 above)
Comment (geoffharcourt) [inline style/android/java/install.sh]: https://en.wikipedia.org/wiki/Sisyphus

File: style/android/java/install.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+         $HOME/Library/Preferences/IdeaIC*        \
+         $HOME/Library/Preferences/AndroidStudio* \
+         $HOME/.IntelliJIdea*/config              \
+         $HOME/.IdeaIC*/config                    \
+         $HOME/.AndroidStudio*/config
+do
+  if [[ -d $i ]]; then
```
Comment (gabebw) [inline style/android/java/install.sh]: I'd quote `$i` everywhere: `-d "$i"`, `"$i/codestyles"`, etc

File: style/android/java/README.md (hunk 1 above)
Comment (gabebw) [inline style/android/java/README.md]: lowercase thoughtbot

File: style/android/java/install.sh (hunk 4 above)
Comment (gabebw) [inline style/android/java/install.sh]: Also, maybe `directory` instead of `i`?

File: style/android/java/configs/codestyles/ThoughtbotAndroid.xml [hunk 6]
```diff
@@ -0,0 +1,384 @@
+<code_scheme name="ThoughtbotAndroid">
```
Comment (mike-burns) [inline style/android/java/configs/codestyles/ThoughtbotAndroid.xml]: I'm going to sound crazy here but, can we lowercase this 'T' too?

File: style/android/java/configs/codestyles/ThoughtbotAndroid.xml [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+    <value />
+  </option>
+  <option name="IMPORT_LAYOUT_TABLE">
+    <value>
+      <package name="" withSubpackages="true" static="false" />
+      <emptyLine />
+      <package name="" withSubpackages="true" static="true" />
+    </value>
+  </option>
+  <option name="RIGHT_MARGIN" value="100" />
```
Comment (mike-burns) [inline style/android/java/configs/codestyles/ThoughtbotAndroid.xml]: !

File: style/android/java/install.bat [hunk 8]
```diff
@@ -0,0 +1,22 @@
+REM Installs Thoughtbot's IntelliJ configs into your user configs.
```
Comment (mike-burns) [inline style/android/java/install.bat]: thoughtbot

File: style/android/java/install.bat [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+echo Installing Thoughtbot IntelliJ configs...
+
+setlocal enableDelayedExpansion
+
+for /D %%i in (%userprofile%\.AndroidStudio*) do call :copy_config %%i
+for /D %%i in (%userprofile%\.IdeaIC*) do call :copy_config %%i
+for /D %%i in (%userprofile%\.IntelliJIdea*) do call :copy_config %%i
+
+echo.
+echo Restart IntelliJ and/or AndroidStudio, go to preferences, and apply 'ThoughtbotAndroid'.
```
Comment (mike-burns) [inline style/android/java/install.bat]: As a member of the  "And/or"  Hater Club, I must request that you go with "or" instead.

File: style/android/java/install.sh [hunk 10]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (mike-burns) [inline style/android/java/install.sh]: `#!/usr/bin/env bash` , though it also looks like you could make this into a shell script by replacing `BASH_SOURCE[0]` with `$0` and replacing `[[` with `[`. If you do turn this into a shell script, `#!/bin/sh` is appropriate instead.

File: style/android/java/install.sh [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    cp -frv "$CONFIGS/codestyles"/* $i/codestyles
+
+    # Install inspections
+    mkdir -p $i/inspection
+    cp -frv "$CONFIGS/inspection"/* $i/inspection
+  fi
+done
+
+echo "Done."
+echo ""
```
Comment (gabebw) [inline style/android/java/install.sh]: I think this can be `echo`

File: style/android/java/install.bat (hunk 8 above)
Comment (teoljungberg) [inline style/android/java/install.bat]: I feel like I have to ask, do we have any windows users at the company?

If not, should we skip this `.bat` file?

**Pull request merged**